qiskit/qpy/binary_io/circuits.py

```diff
@@ -239,6 +261,14 @@ def _read_instruction(file_obj, circuit, registers, custom_instructions, version
 
     if gate_name in {"IfElseOp", "WhileLoopOp"}:
         gate = gate_class(condition_tuple, *params)
+    elif version >= 5 and issubclass(gate_class, ControlledGate):
+        if gate_name == "MCPhaseGate":
```

Comment:
I found `MCU1Gate` takes `num_ctrl_qubits` as a required argument. We don't need to add the gate here? Also I found several `MC*` gates in `qiskit/circuit/library/standard_gates/__init__.py` that doesn't take any `*params` in the constructor. Since params is a variadic argument, it's probably fine but have you tried serialization of these gates?

Reply:
Let me adds some tests for MCU1Gate and some other controlled gates and I'll fix issues if they come up

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8055/commits/354cf8562c5ba82046a717c51e5ddfdb132701f5